packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,254 @@
+
+    const current = getOrElse(amountByAddress, address, () => empty);
+    const target = add(current, lienDelta);
+
+    // Since our mint is not subject to normal supply/demand,
+    // minting here and failing in setLiened below does no harm.
+    const attestationPaymentP = mintZCFMintPayment(
+      zcf,
+      zcfMint,
+      wrapLienedAmount(address, lienDelta),
+    );
+
+    // If setLiened returns successfully, then
+    // the preconditions were indeed met and we
+    // can update the stored liened amount.
+    await E(lienBridge).setLiened(address, current, target);
+    amountByAddress.set(address, target);
+    return attestationPaymentP;
+  };
+
+  /** @param {ZCFSeat} seat */
+  const returnAttestation = async seat => {
+    assertProposalShape(seat, { give: { Attestation: null }, want: {} });
+
+    /** @type {Amount<'copyBag'>} */
+    const attestationAmount = seat.getAmountAllocated(KW.Attestation);
+    const { address, lienedAmount: amountReturned } =
+      unwrapLienedAmount(attestationAmount);
+
+    const liened = amountByAddress.get(address); // or throw
+    const updated = subtract(liened, amountReturned); // or throw
+
+    await E(lienBridge).setLiened(address, liened, updated);
+    // COMMIT. Burn the escrowed attestation payment and exit the seat
+    amountByAddress.set(address, updated);
+    zcfMint.burnLosses(harden({ Attestation: attestationAmount }), seat);
+    seat.exit();
+  };
+
+  /**
+   * Get liened amount synchronously.
+   *
+   * @param {string} address
+   * @param {Brand<'nat'>} brand
+   */
+  const getLiened = (address, brand) => {
+    assert.typeof(address, 'string');
+    assert.equal(
+      brand,
+      stakeBrand,
+      X`This contract can only make attestations for ${brand}`,
+    );
+    return getOrElse(amountByAddress, address, () => empty);
+  };
+
+  const lienMint = harden({
+    mintAttestation,
+    returnAttestation,
+    getLiened,
+    /**
+     * @param { Amount<'copyBag'> } attAmt
+     * @throws { Error } if `attAmt` payload length is not 1
+     */
+    unwrapLienedAmount: attAmt => unwrapLienedAmount(attAmt).lienedAmount,
+  });
+  return harden({ issuer, brand: attBrand, lienMint });
+};
+
+/**
+ * Authorize each account holder to lien some of their staked assets
+ * and get an attestation that the lien is in place.
+ *
+ * @param {ZCF} zcf
+ * @param {Brand} stakeBrand
+ * @param {ERef<StakingAuthority>} lienBridge
+ *
+ * NOTE: the liened amount is kept both here in JS and on the
+ * other side of the lienBridge. The mint created here
+ * is the authority on the liened amount. The liened amount
+ * on the other side of the bridge may lag. Do not create two
+ * mints that share a bridge.
+ *
+ * The provideAttestationMaker method on the returned creatorFacet
+ * provides the authorization for an address. The account holder
+ * can then use attMaker.makeAttestation(lienDelta).
+ *
+ * Note the Attestation keyword for use in returnAttestation offers.
```

Comment:
is this comment for an earlier version? i'm not seeing any keywords in this function. 

I do see that `returnAttestation` itself has that keyword

Reply:
This JSDoc comment is meant to specify what clients would want to know about this contract. I figure keywords are part of that, even though they only get used in `returnAttestation`.

Reply:
Agreed. Though if we use the bare string `Attestation` we're countering some of the benefits of it being defined in the code by `KW.Attestation` instead of simply `'Attestation'`. If the goal of defining the manifest constant is that we can easily change the underlying value, then I think this comment should use it. If not, then I think we should reconsider whether the manifest constants are paying for their way.

Separately, as a reader I still don't get what I'm supposed to note. Maybe "Note that `Attestation` is the keyword is in `returnAttestation` offers."